docs/_includes/toggle-button.html

```diff
@@ -0,0 +1,480 @@
+                    </span>
+                </span>
+                <span class="toggle-button__content">
+                    <span class="toggle-button__title">CSS Image Button 2</span>
+                    <span class="toggle-button__subtitle">With cover background size</span>
+                </span>
+            </button>
+        </div>
+    </div>
+
+    {% highlight html %}
+<button type="button" class="toggle-button" aria-pressed="false">
+    <span class="toggle-button__image-container">
+        <span class="toggle-button__image">
+            <img src="{{ page.static_dir }}/img/tb-profile-pic.jpg" alt="">
+        </span>
+    </span>
+    <span class="toggle-button__content">
+        <span class="toggle-button__title">Image Button Title</span>
+        <span class="toggle-button__subtitle">Image Button Subtitle</span>
+    </span>
+</button>
+    {% endhighlight %}
+
+    <h3>Minimal View Theme</h3>
+
+    <p>The Minimal View Theme toggle buttons should be used when the contents are single-line and very short, as in shoe sizing, for example. This theme is geared more towards allowing the display of many more options and maximize screen real estate.</p>
+
+    <p>In an individual toggle button, you may use this theme by adding the <span class="highlight">toggle-button-theme-minimal</span> modifier to the toggle button.</p>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <button type="button" class="toggle-button toggle-button-theme-minimal" aria-pressed="false">
+                <span class="toggle-button__content">
+                    <span class="toggle-button__title">4.5</span>
+                </span>
+            </button>
+        </div>
+    </div>
+
+    {% highlight html %}
+<button type="button" class="toggle-button toggle-button-theme-minimal" aria-pressed="false">
+    <span class="toggle-button__content">
+        <span class="toggle-button__title">4.5</span>
+    </span>
+</button>
+    {% endhighlight %}
+
+    <h3>Gallery View Theme</h3>
+
+    <p>The Gallery View Theme toggle buttons should be used when there will be fewer buttons (as larger buttons take up more screen real estate) and it's desired to have a more vertical/square implementations with more emphasis on graphical elements (images/icons).</p>
+
+    <p>In an individual toggle button, you may use this theme by adding the <span class="highlight">toggle-button-theme-gallery</span> modifier to the toggle button.</p>
+
+    <h4>Toggle Button with Image</h4>
+
+    <p>To add an image to a toggle button, you can use the <span class="highlight">toggle-button__image-container</span> block.</p>
+
+    <div class="demo">
+        <div class="demo__inner">
+            <button type="button" class="toggle-button toggle-button-theme-gallery" aria-pressed="false">
```

Comment:
How about a modifier instead of a new block? e.g. `toggle-button--gallery-layout` instead of `toggle-button-theme-gallery`? or are there some possible child blocks of `toggle-button-theme-gallery`?

Reply:
Good catch. No, that was a slip up on my part. I'll change them to modifiers. But first, let's discuss the naming... 

I don't mind using "layout" instead of "theme" either. I used "theme" for the reason mentioned in the previous comment - to scale a bit better when, in the future, we are able to responsively go from list view to gallery view (and vice versa) based on viewport/container width. I'm imagining the slight awkwardness of seeing a 'list view _layout_' button named 'gallery view _layout_'. I suppose the same can be said for using "theme." It just seemed less expressive of the actual layout; it's slightly more abstract. There's going to be some weirdness no matter what. Just trying to keep it to a minimum.

Reply:
> I'm imagining the slight awkwardness of seeing a 'list view layout' button named 'gallery view layout'.
>> Would this ever be the case? My understanding was that adding an explicit class for list or gallery would force the component to always use that layout, while leaving it out would result in responsive switching

That approach is an option, but when we do get around to that iteration, my idea was to have a distinct additional modifier, maybe something like `toggle-button--gallery-layout-only` or `toggle-button--gallery-layout-immutable`. I think being explicit is better as it avoids additional complications with possible other layout themes being added in the future create implied fallback confusion.

Reply:
I think our decision here is going to be affected by my previous comment:

> It might result in some duplication across list view and gallery, in which case perhaps there's a common class (or mixin) that they can both use?

Reply:
We could also create a concept of "sets" that would allow transitions from one theme to another within the "set" only, but that decision is not something we have to make now; that's more of a `toggle-button-group` thing. For now,  having a naming convention that could fit better regardless of approach would be good.

Reply:
That makes sense, I think if the goal is simplicity then the best may just be `toggle-button--gallery` and `toggle-button--list`.

Reply:
I think the brevity makes a lot of sense in the short term, but if we ever potentially add more modifiers for some other purpose, it might muddy the water as it might make it difficult to tell which modifiers are for what purpose. I think being slightly more verbose now and being explicit about the role of the modifiers avoids that potential pitfall. Doing a cost/benefit here, I would opt to just play it safe and have the classes a bit longer.